proposals/nnnn-cloud-app-transport-adapter.md

```diff
@@ -0,0 +1,243 @@
+        <param name="watchdog_timer_ms" type="Integer" minvalue="1"
+            maxvalue="65225" mandatory="false"/>
+        <param name="cloud_app_endpoint", type="String", minlength="0" maxlength="255", mandatory="false">
+        <param name="certificate" type="String" minlength="0" maxlength="255"
+            mandatory="false" />
+    </struct>
+```
+This feature should be supported by both regular and external policy build configurations.
+
+### SDL Server
+
+SDL Server must create an interface for managing additional policy data. Interface should allow for adding, updating, and revoking endpoint urls and SSL certificates. This work should be minimal since its only two extra parameters.
+
+## Alternatives considered
+
+Since there are many aspects to this proposal, I have considered a few alternatives.
+
+### Using Tokens to authenticate
+
+1. Core gets web app endpoints and associated tokens from policy server after a policy table update. 
+2. Core opens a websocket connection for each webapp endpoint located in the policy table. 
+3. Core sends the VIN and token to the application and waits for the application to respond with data.
+4. The web application server sends a request to the policy server to identify the incoming connection via the received VIN and token. 
+5. If the policy server responds matches that the VIN and token are valid it responds with success to the web app server. 
+6. The web app server starts an RPC service with Core.
+
+```JSON
+        "app_policies": {
+            "default": {
+                "keep_context": false,
+                "steal_focus": false,
+                "priority": "NONE",
+                "default_hmi": "NONE",
+                "groups": ["Base-4"]
+            },
+            "IOSApp12345": {
+                "keep_context": true,
+                "steal_focus": true,
+                "priority": "NONE",
+                "default_hmi": "NONE",
+                "groups": ["Base-4", "Location-1", "Emergency-1"]
+            },
+            "WebAppID12345" : {
+                "keep_context": true,
+                "steal_focus": true,
+                "priority": "NONE",
+                "default_hmi": "NONE",
+                "groups": ["Base-4"],
++                "endpoint": "https://10.0.0.1:8080",
++                "certificate" : "-----BEGIN CERTIFICATE-----\n" ...,
++                "token": "1234"               
+            }
+        }
+```
+
+Using a token to authenticate means that no RPC service would be opened with unknown users.
+
+### Using the Mobile Device to Connect
+
+If future vehicles will not have dedicated LTE connections, then the mobile device could be used as the proxy for connecting web apps to core. The phone would connect to core using traditional SDL transports, and then also maintain a socket connection with the cloud application. Any data sent from Core/Cloud App would be passed through the phone. 
```

Comment:
Please capitalize "Core" throughout to match formatting within the rest of the proposal.